### DISCUSSION TITLE
Add more skins

### DISCUSSION THREAD

**Discussion by waltiti17** [Builty/TonexOneController#13] [General]
there is an easy way to add skins? thanks

**Comment by Builty** [comment]
Its currently not easy. I wanted to have them on SD card but the Waveshare 4.3" board has the SD card chip select line on an I2C IO expander, which is a shared bus with the touch controller. Really hard to avoid conflics/deadlocks that way.
I also tried to copy all images to ram on boot but ram is not big enough for that.

So currently the procedure is:
- get the skin image
- edit in photoshop to set the size and remove the background. Save as PNG
- add the PNG to Squareline Studio project as an asset
- create an image object like the rest, hidden by default
- build the Squareline project
- this generates a new .c file in source/main/ui_generated/images. This file needs to be added to source/main/CmakeLists.txt
- add the new skin to the enum in control.h
- update display.c init_amp_skins() to add the new skin
- build project

I tried to find all the common skins. If I've missed some you can let me know and I can add them.

**Comment by waltiti17** [comment]
Thanks, I have 2 tonex one, and I using one of them for overdrives, so if its have the skins for most common pedals will be great, like the IK butique pedals , Tube screamer, OCD ,horizon Precision drive, centaur, red llama, rat, ds1, maxon, metal zone, mt2,  prince of tone, EP Booster, king of tone, 

Witing for the ESP32-S3 4.3inch LCD  be delivered and try it. 

thanks

**Reply by Builty** [reply]
OK I understand. I will see what I can do.

**Comment by Builty** [comment]
I have committed changes to add pedal skins now. In menu config you can select if you want pedal skins or amp skins.
I can't fit both at once due to ram limit (ESP32 copies constant data to PSRAM on boot. Disabling this results in poor performance)

**Comment by Builty** [comment]
code done and committed
